app/src/org/commcare/dalvik/activities/HomeButtons.java

```diff
@@ -0,0 +1,212 @@
+                        R.drawable.home_saved,
+                        R.color.cc_light_cool_accent_color,
+                        getViewOldFormsListener(activity)),
+                HomeCardDisplayData.homeCardDataWithDynamicText(Localization.get("home.forms.incomplete"), R.color.white,
+                        R.drawable.home_incomplete,
+                        R.color.solid_dark_orange,
+                        getIncompleteButtonListener(activity),
+                        getIncompleteButtonNotificationText(activity)),
+                HomeCardDisplayData.homeCardDataWithNotification(Localization.get(syncKey), R.color.white,
+                        R.color.white,
+                        R.drawable.home_sync,
+                        R.color.cc_brand_color,
+                        R.color.cc_brand_text,
+                        getSyncButtonListener(activity),
+                        getSyncButtonNotificationText(activity)),
+                HomeCardDisplayData.homeCardDataWithStaticText(Localization.get("home.report"), R.color.white,
+                        R.drawable.home_report, R.color.cc_attention_negative_color,
+                        getReportButtonListener(activity)),
+                HomeCardDisplayData.homeCardDataWithNotification(Localization.get(logoutMessageKey), R.color.white,
+                        R.color.white,
+                        R.drawable.home_logout, R.color.cc_neutral_color, R.color.cc_neutral_text,
+                        getLogoutButtonListener(activity),
+                        getLogoutButtonNotificationText(activity)),
+        };
+
+        return getVisibleButtons(allButtons, buttonsToHide);
+    }
+
+    private static HomeCardDisplayData[] getVisibleButtons(HomeCardDisplayData[] allButtons,
+                                                           Vector<String> buttonsToHide) {
+        int visibleButtonCount = buttonNames.length - buttonsToHide.size();
+        HomeCardDisplayData[] buttons = new HomeCardDisplayData[visibleButtonCount];
+        int visibleIndex = 0;
+        for (int i = 0; i < buttonNames.length; i++) {
+            if (!buttonsToHide.contains(buttonNames[i])) {
+                buttons[visibleIndex] = allButtons[i];
+                visibleIndex++;
+            }
+        }
+        return buttons;
+    }
+
+    private static View.OnClickListener getViewOldFormsListener(final CommCareHomeActivity activity) {
+        return new View.OnClickListener() {
+            public void onClick(View v) {
+                activity.goToFormArchive(false);
+            }
+        };
+    }
+
+    private static View.OnClickListener getSyncButtonListener(final CommCareHomeActivity activity) {
+        return new View.OnClickListener() {
+            public void onClick(View v) {
+                activity.syncButtonPressed();
+            }
+        };
+    }
+
+    private static TextSetter getSyncButtonNotificationText(final CommCareHomeActivity activity) {
```

Comment:
I'm finding the names of these 3 methods (this and the other 2 get____NotificationText() methods) confusing, given that they return a TextSetter for all of the text for 1 home card, but it sounds like it would just be returning a String (the notification text).